src/commands/Matcha.ts

```diff
@@ -104,17 +107,31 @@ export default class Matcha extends Command {
     const shuffledMembersList = Matcha.shuffle(users);
     const memberPairings = [];
 
+    const groupsize = interaction.options.getInteger('groupsize', true);
+    Logger.error(`/matcha - ready to match ${shuffledMembersList}`);
     while (shuffledMembersList.length > 0) {
       let pairedMembers: GuildMember[];
-      if (shuffledMembersList.length % 2 === 1) {
-        // If there's an odd number of people, we start with a group of 3 to correct it.
-        pairedMembers = shuffledMembersList.splice(0, 3);
+      const extraMembers = shuffledMembersList.length % groupsize;
+      const numGroups = Math.floor(shuffledMembersList.length / groupsize);
+      Logger.error(`/matcha - numGroups ${numGroups}, extraMembers: ${extraMembers} groupsize ${groupsize}`);
+      if (extraMembers !== 0) {
+        // If length % size is off add people to the first group or first n groups
+        // for one group remaining add all extras
+        if (numGroups === 1) {
+          pairedMembers = shuffledMembersList.splice(0, shuffledMembersList.length);
+          Logger.error(`/matcha - pair ${pairedMembers}`);
+        }
+        // otherwise, disperse extra people across groupss
+        else {
+          const addedToGroup = Math.floor(extraMembers / numGroups);
+          pairedMembers = shuffledMembersList.splice(0, groupsize + addedToGroup);
+        }
```

Comment:
Nit: I think we can make this code more clear by breaking the code out into a helper function (maybe `splitIntoGroups()`?)

Additionally, I think we need to consider edge cases here:
- What if the group size is 3 and we have 5 total members? I think this code makes a pair of 4 and leaves the last person lonely